### PR TITLE
FIX(scss) disabled states for interactive elements

### PR DESCRIPTION
**Pull request by vvalentin-lucca**
Disabled states for:

- [x] Button,
- [x] Textfield,
- [x] Checkbox,
- [x] Radio,
- [x] Switch,
- [x] RadioButtons,
- [x] Framed.

-----

Capture d’écran 2021-11-19 à 15 43 29

image

image

image

image

image

image

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by AlexiaAlarza** [comment]
other branches to merge with this one : 
- https://github.com/LuccaSA/lucca-front/pull/1479

**Comment by RipripKipkip** [comment]
other branches to merge with this one : 
- https://github.com/LuccaSA/lucca-front/pull/1480

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Review by AlexiaAlarza — APPROVED** [review]

File: packages/scss/src/components/_button.scss [hunk 1]
```diff
@@ -5,63 +5,62 @@
...
+		}
 
-	&:active:not([disabled]) {
-		background-color: _color($palette, "200");
-		color: _color($palette);
-		box-shadow: 0 2px 8px rgba(19, 29, 53, .04), 0 1px 2px rgba(19, 29, 53, .06), 0 0 0 4px _color($palette, "200"), inset 0 0 0 1px _color($palette, "500");
+		&:focus {
+			background-color: _color($palette, "100");
+			color: _color($palette);
+			box-shadow: 0 2px 8px rgba(19, 29, 53, .04), 0 1px 2px rgba(19, 29, 53, .06), 0 0 0 4px _color($palette, "100"), inset 0 0 0 1px _color($palette, "400");
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/_button.scss:55]:
```suggestion
			box-shadow: _theme('commons.elevations.elevation-1'), 0 0 0 4px _color($palette, "100"), inset 0 0 0 1px _color($palette, "400");
```

File: packages/scss/src/components/_button.scss [hunk 2]
```diff
@@ -5,63 +5,62 @@
...
+			outline: none;
+		}
+
+		&:active {
+			background-color: _color($palette, "200");
+			color: _color($palette);
+			box-shadow: 0 2px 8px rgba(19, 29, 53, .04), 0 1px 2px rgba(19, 29, 53, .06), 0 0 0 4px _color($palette, "200"), inset 0 0 0 1px _color($palette, "500");
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/_button.scss:62]:
```suggestion
			box-shadow: _theme('commons.elevations.elevation-1'), 0 0 0 4px _color($palette, "200"), inset 0 0 0 1px _color($palette, "500");
```

File: packages/scss/src/components/_button.scss [hunk 3]
```diff
@@ -220,37 +219,41 @@
...
-	
-		&:active:not([disabled]) {
-			background-color: _color("grey", "200");
-			color: _color("text.default");
-			box-shadow: 0 2px 8px rgba(19, 29, 53, .04), 0 1px 2px rgba(19, 29, 53, .06), 0 0 0 4px _color("primary", "200"), inset 0 0 0 1px _color("primary", "500");
+		&:not([disabled], .is-disabled) {
+			&:hover {
+				background-color: _color("grey", "100");
+				color: _color("text.default");
+				box-shadow: 0 2px 8px rgba(19, 29, 53, .04), 0 1px 2px rgba(19, 29, 53, .06), inset 0 0 0 1px _color("grey", "400");
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/_button.scss:232]:
```suggestion
				box-shadow: _theme('commons.elevations.elevation-1'), inset 0 0 0 1px _color("grey", "400");
```

File: packages/scss/src/components/_button.scss [hunk 4]
```diff
@@ -220,37 +219,41 @@
...
+			}
+
+			&:focus {
+				background-color: _color("grey", "100");
+				color: _color("text.default");
+				box-shadow: 0 2px 8px rgba(19, 29, 53, .04), 0 1px 2px rgba(19, 29, 53, .06), 0 0 0 4px _color("primary", "100"), inset 0 0 0 1px _color("primary", "400");
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/_button.scss:238]:
```suggestion
				box-shadow: _theme('commons.elevations.elevation-1'), 0 0 0 4px _color("primary", "100"), inset 0 0 0 1px _color("primary", "400");
```

File: packages/scss/src/components/_button.scss [hunk 5]
```diff
@@ -220,37 +219,41 @@
...
+				outline: none;
+			}
+
+			&:active {
+				background-color: _color("grey", "200");
+				color: _color("text.default");
+				box-shadow: 0 2px 8px rgba(19, 29, 53, .04), 0 1px 2px rgba(19, 29, 53, .06), 0 0 0 4px _color("primary", "200"), inset 0 0 0 1px _color("primary", "500");
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/_button.scss:245]:
```suggestion
				box-shadow: _theme('commons.elevations.elevation-1'), 0 0 0 4px _color("primary", "200"), inset 0 0 0 1px _color("primary", "500");
```

File: packages/scss/src/components/_button.scss [hunk 6]
```diff
@@ -385,15 +408,18 @@
...
 		border-radius: 0;
 		display: block;
 		margin: 0;
 		padding-left: _theme("spacings.small");
 		padding-right: _theme("spacings.small");
 		position: relative;
-		border-right: 1px solid rgba(0, 0, 0, .1);
+
+		&:not(.mod-outline, .mod-outlined) {
+			border-right: 1px solid rgba(0, 0, 0, .1);
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/_button.scss:421]:
```suggestion
			border-right: _theme('commons.divider.width') solid rgba(0, 0, 0, .1);
```

**Review by AlexiaAlarza — APPROVED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1478

**Review by AlexiaAlarza — APPROVED** [review]

**Pull request merged**